File: schemas/marketplace/mongodb/vendor_addresses.js
// vendor_addresses: Physical addresses associated with vendors.
// See README.md for full design rationale.

const mongoose = require("mongoose");

const vendorAddressSchema = new mongoose.Schema(
  {
    vendor_id: { type: mongoose.Schema.Types.ObjectId, ref: "Vendor", required: true },
    type: {
      type: String,
      enum: ["business", "warehouse", "return"],
      required: true,
    },
    label: { type: String, default: null },
    address_line1: { type: String, required: true },
    address_line2: { type: String, default: null },
    city: { type: String, required: true },
    region: { type: String, default: null },
    postal_code: { type: String, default: null },
    country: { type: String, required: true },
    phone: { type: String, default: null },
    is_default: { type: Boolean, required: true, default: false },
  },
  { timestamps: { createdAt: "created_at", updatedAt: "updated_at" } }
);

vendorAddressSchema.index({ vendor_id: 1, type: 1 });

module.exports = mongoose.model("VendorAddress", vendorAddressSchema);

File: schemas/marketplace/spacetimedb/vendor_addresses.rs
// vendor_addresses: Physical addresses associated with vendor operations.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[derive(SpacetimeType, Clone)]
pub enum VendorAddressType {
    Business,
    Warehouse,
    Return,
}

#[spacetimedb::table(name = vendor_addresses, public)]
pub struct VendorAddress {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub vendor_id: String, // UUID, FK -> vendors.id (cascade delete)

    pub address_type: VendorAddressType,
    pub label: Option<String>,
    pub address_line1: String,
    pub address_line2: Option<String>,
    pub city: String,
    pub region: Option<String>,
    pub postal_code: Option<String>,
    pub country: String,
    pub phone: Option<String>,
    pub is_default: bool,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
